lib/terraform/runner.rb

```diff
@@ -9,32 +9,59 @@ class << self
       def available?
         return @available if defined?(@available)
 
-        response = terraform_runner_client.get('ping')
-        @available = response.status == 200
+        response = terraform_runner_client.get('live')
+        @available = response.status == 200 && JSON.parse(response.body)['status'] == 'UP'
       rescue
         @available = false
       end
 
       # Run a template, initiates terraform-runner job for running a template, via terraform-runner api
       #
       # @param input_vars [Hash] Hash with key/value pairs that will be passed as input variables to the
-      #        terraform-runner run
-      # @param template_path [String] Path to the template we will want to run
+      #        terraform-runner run, (execpt :miq_action && :miq_terraform_runner_stack_id used for 'Retirement')
+      #        Note: To run Retirement(delete stack) required vars:
+      #              - :miq_action=>'Retirement'
+      #              - :miq_terraform_runner_stack_id=>{{stack_id-from-terraform-runner}}
       # @param tags [Hash] Hash with key/values pairs that will be passed as tags to the terraform-runner run
       # @param credentials [Array] List of Authentication objects to provide to the terraform run
       # @param env_vars [Hash] Hash with key/value pairs that will be passed as environment variables to the
       #        terraform-runner run
       # @return [Terraform::Runner::ResponseAsync] Response object of terraform-runner create action
       def run_async(input_vars, template_path, tags: nil, credentials: [], env_vars: {})
-        _log.debug("Run_aysnc template: #{template_path}")
-        response = create_stack_job(
-          template_path,
-          :input_vars  => input_vars,
-          :tags        => tags,
-          :credentials => credentials,
-          :env_vars    => env_vars
-        )
-        Terraform::Runner::ResponseAsync.new(response.stack_id)
+        action   = input_vars[:miq_action]
+        stack_id = input_vars[:miq_terraform_runner_stack_id]
+        input_vars.delete(:miq_action)
+        input_vars.delete(:miq_terraform_runner_stack_id)
+
+        case action
+        when ResourceAction::RETIREMENT
```

Comment:
This doesn't seem right, I expected that we'd expose the create_stack and delete_stack on Terraform::Runner rather than using some hash value in one of the arguments to completely change the behavior of the method call.

Reply:
Originally, my thoughts were also this, but this made things easier

Reply:
done